Sources/TuistCore/Simulator/SimulatorController.swift

```diff
@@ -178,7 +180,7 @@ public final class SimulatorController: SimulatorControlling {
         case .iOS: platform = .iOS
         case .watchOS: platform = .watchOS
         case .tvOS: platform = .tvOS
-        case .macOS: return "platform=OS X,arch=x86_64"
+        case .macOS: return "generic/platform=macOS,name=Any Mac"
```

Comment:
I think we still want to specify the `arch`, or what do you think?

Reply:
Can't we just specify the correct arch, depending on the current one?

Something like
```
    private func platformArch() -> String {
        switch DeveloperEnvironment.shared.architecture {
        case .arm64:
            return "arm64"
        case .x8664:
            return "x86_64"
        }
    }
```

Reply:
Yes that's one approach, although i think it would be better to somehow allow this to be specified. Some projects can't be built on arm and some can't be built on just intel, not sure how to strike a balance here

Reply:
I see, but I would at least fix the problem we have (i.e. merging this PR), and then think about how to fix that other one, what do you think?

Reply:
Yes I agree, I don't have time right now to take a look but feel free to update this. If not i can try to get to it soon

Reply:
Updated the code to what I have proposed, let me know what do you think